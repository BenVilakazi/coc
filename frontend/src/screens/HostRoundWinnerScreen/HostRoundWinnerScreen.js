import React, { useContext, useEffect } from 'react';
import styled from 'styled-components';
import HostLayout from '../../layouts/HostLayout';
import { HostContext } from '../../contexts/HostContext/HostContext';
import PlayerList from '../../components/PlayerList/PlayerList';
import InGameSettingsModal from '../../components/HostSettingsMenu/InGameSettingsModal.js';
import JoinCode from '../../components/JoinCode/JoinCode';
import BlackCard from '../../components/Cards/BlackCard';
import WhiteCard from '../../components/Cards/WhiteCard';

const LeftPanelWrapper = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;

  .player-list-wrapper {
    overflow-x: auto;
    flex-grow: 1;
  }

  .buttons-wrapper {
    display: block;
    margin-bottom: 20px;
  }

  .bottom-left-wrapper {
    margin: 20px auto 32px;
    position: relative;
    bottom: 0;
    left: 0;
    right: 0;
  }

  .join-code-wrapper {
    display: flex;
    justify-content: center;
  }

  .host-pregame-button {
    padding: 0;
    background-color: var(--primary-background-color);
    color: var(--primary-text-color);
    margin: auto auto 12px;
  }
`;

const RightPanelWrapper = styled.div`
  flex: 1;

  display: flex;
  flex-direction: column;
  align-items: center;

  .winner-display {
    display: flex;
    flex-direction: column;
    margin: 25px 0;
    line-height: 1em;

    font-size: 56px;
    font-weight: 900;
    text-transform: uppercase;

    span {
      font-size: 24px;
      text-indent: -20px;
      margin-bottom: -8px;
      line-height: 1.5rem;
    }
  }
`;

const WhiteCardWrapper = styled.div`
  display: grid;
  grid-template-columns: ${(props) => `repeat(${props.columns}, auto)`};
  gap: 20px;
  place-items: center;
  justify-content: center;
  padding: 40px 0;
  width: 100%;
  max-height: 250px;
`;

function LeftPanel() {
  const { state } = useContext(HostContext);
  const { lobbyID } = state;

  return (
    <LeftPanelWrapper>
      <div className="player-list-wrapper">
        <PlayerList />
      </div>
      <div className="bottom-left-wrapper">
        <div className="join-code-wrapper">
          <JoinCode code={lobbyID} />
        </div>
      </div>
    </LeftPanelWrapper>
  );
}

function RightPanel() {
  const { state } = useContext(HostContext);

  const { selectedBlackCard, players, czarSelection } = state;

  const winner = players[czarSelection];

  return (
    <RightPanelWrapper columns={winner.submittedCards.length}>
      <h1 className="winner-display" data-testid="winner-display">
        <span>WINNER:</span>
        {winner.name.toUpperCase()}
      </h1>

      <BlackCard
        pickCount={selectedBlackCard.pick}
        data-test-id="black-card"
        winnerScreen
      >
        {selectedBlackCard.text}
      </BlackCard>

      <WhiteCardWrapper columns={winner.submittedCards.length}>
        {winner.submittedCards.map((card) => (
          <WhiteCard key={card}>
            {/* When round changes, the card may become undefined temporarily resulting in an error */}
            {winner.cards[card] && winner.cards[card].text}
          </WhiteCard>
        ))}
      </WhiteCardWrapper>
    </RightPanelWrapper>
  );
}

function startNextRound(dispatch) {
  dispatch({ type: 'ADD_PLAYERS_FROM_STAGING', payload: {} });

  dispatch({ type: 'REMOVE_SUBMITTED_CARDS_FROM_PLAYER', payload: {} });

  dispatch({ type: 'SET_NEXT_CZAR', payload: {} });

  dispatch({ type: 'SELECT_BLACK_CARD', payload: {} });

  dispatch({ type: 'DEAL_WHITE_CARDS', payload: {} });
}

function endGame(dispatch, payload) {
  dispatch({ type: 'GAME_OVER', payload });
}

function HostRoundWinnerScreen() {
  const { state, dispatch } = useContext(HostContext);
  const { winnerScreenDisplayTime, winningScore } = state.gameSettings;
  const { players, playerIDs, czarSelection } = state;

  useEffect(() => {
    const czar = playerIDs.find((id) => players[id].isCzar);
    const losers = playerIDs.filter(
      (playerID) => !players[playerID].isCzar && playerID !== czarSelection,
    );

    dispatch({
      type: 'SEND_END_OF_ROUND_MESSAGES',
      payload: {
        winnerName: players[czarSelection].name,
        winnerId: czarSelection,
        losers,
        czar,
      },
    });

    setTimeout(() => {
      if (players[czarSelection].score >= winningScore) {
        return endGame(dispatch, { gameWinner: czarSelection, playerIDs });
      }

      return startNextRound(dispatch);
    }, winnerScreenDisplayTime);
  }, []);

  return (
    <HostLayout
      className="primary-background"
      left={<LeftPanel />}
      right={<RightPanel />}
      modal={<InGameSettingsModal />}
    />
  );
}

export default HostRoundWinnerScreen;
