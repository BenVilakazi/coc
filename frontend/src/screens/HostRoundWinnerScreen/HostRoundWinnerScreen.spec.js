import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { HostContext } from '../../contexts/HostContext/HostContext';
import HostRoundWinnerScreen from './HostRoundWinnerScreen';

describe('Host Winner Screen', () => {
  describe('rendering', () => {
    const dispatch = jest.fn();
    const state = {
      gameState: 'showing-winning-cards',
      lobbyID: 'ABCD',
      players: {
        ID1: {
          name: 'foo',
          score: 0,
          isCzar: false,
          submittedCards: [0],
          cards: [
            { text: 'aaaa' },
            { text: 'bbbb' },
            { text: 'cccc' },
            { text: 'dddd' },
          ],
        },
        ID2: {
          name: 'bar',
          score: 0,
          isCzar: true,
          submittedCards: [2],
          cards: [
            { text: 'aaaa' },
            { text: 'bbbb' },
            { text: 'cccc' },
            { text: 'dddd' },
          ],
        },
        ID3: {
          name: 'baz',
          score: 0,
          isCzar: false,
          submittedCards: [1],
          cards: [
            { text: 'eeee' },
            { text: 'ffff' },
            { text: 'gggg' },
            { text: 'hhhh' },
          ],
        },
      },
      selectedBlackCard: {
        text: 'Test Black Card',
        pick: 1,
        pack: 0,
      },
      playerIDs: ['ID1', 'ID2', 'ID3'],
      gameSettings: { maxPlayers: 8, winningScore: 7, selectedPacks: [] },
      czarSelection: 'ID1',
    };

    it('renders', () => {
      const tree = renderer
        .create(
          <HostContext.Provider value={{ state, dispatch }}>
            <HostRoundWinnerScreen />
          </HostContext.Provider>,
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('renders the winning card', () => {
      render(
        <HostContext.Provider value={{ state, dispatch }}>
          <HostRoundWinnerScreen />
        </HostContext.Provider>,
      );

      expect(screen.queryByText('aaaa')).toBeInTheDocument();
    });

    it('renders the black card', () => {
      render(
        <HostContext.Provider value={{ state, dispatch }}>
          <HostRoundWinnerScreen />
        </HostContext.Provider>,
      );

      expect(screen.queryByTestId('black-card')).toBeInTheDocument();
    });

    it("renders the winner's name in the czar display", () => {
      render(
        <HostContext.Provider value={{ state, dispatch }}>
          <HostRoundWinnerScreen />
        </HostContext.Provider>,
      );

      expect(screen.queryByTestId('winner-display')).toBeInTheDocument();
      expect(screen.queryByTestId('winner-display')).toHaveTextContent(
        'WINNER:FOO',
      );
    });
  });

  describe('functionality', () => {
    const dispatch = jest.fn();
    const state = {
      gameState: 'showing-winning-cards',
      lobbyID: 'ABCD',
      players: {
        ID1: {
          name: 'foo',
          score: 0,
          isCzar: false,
          submittedCards: [0],
          cards: [
            { text: 'aaaa' },
            { text: 'bbbb' },
            { text: 'cccc' },
            { text: 'dddd' },
          ],
        },
        ID2: {
          name: 'bar',
          score: 0,
          isCzar: true,
          submittedCards: [2],
          cards: [
            { text: 'aaaa' },
            { text: 'bbbb' },
            { text: 'cccc' },
            { text: 'dddd' },
          ],
        },
        ID3: {
          name: 'baz',
          score: 0,
          isCzar: false,
          submittedCards: [1],
          cards: [
            { text: 'eeee' },
            { text: 'ffff' },
            { text: 'gggg' },
            { text: 'hhhh' },
          ],
        },
      },
      selectedBlackCard: {
        text: 'Test Black Card',
        pick: 1,
        pack: 0,
      },
      playerIDs: ['ID1', 'ID2', 'ID3'],
      gameSettings: {
        maxPlayers: 8,
        winningScore: 7,
        selectedPacks: [0],
        handSize: 10,
        winnerScreenDisplayTime: 3000,
      },
      czarSelection: 'ID1',
    };

    it('calls setTimeout once with a function and three second timer', async () => {
      jest.useFakeTimers();

      await render(
        <HostContext.Provider value={{ state, dispatch }}>
          <HostRoundWinnerScreen />
        </HostContext.Provider>,
      );

      expect(setTimeout).toHaveBeenCalledTimes(1);
      expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 3000);
    });

    it('does not send next round dispatches before the setTimeout executes', () => {
      render(
        <HostContext.Provider value={{ state, dispatch }}>
          <HostRoundWinnerScreen />
        </HostContext.Provider>,
      );

      jest.advanceTimersByTime(2000);

      expect(dispatch).not.toHaveBeenCalledWith({
        type: 'REMOVE_SUBMITTED_CARDS_FROM_PLAYER',
        payload: {},
      });

      expect(dispatch).not.toHaveBeenCalledWith({
        type: 'SET_NEXT_CZAR',
        payload: {},
      });

      expect(dispatch).not.toHaveBeenCalledWith({
        type: 'SELECT_BLACK_CARD',
        payload: {},
      });

      expect(dispatch).not.toHaveBeenCalledWith({
        type: 'DEAL_WHITE_CARDS',
        payload: {},
      });

      expect(dispatch).toHaveBeenCalledTimes(1);
    });

    it('sends next round dispatches once the setTimeout executes', () => {
      render(
        <HostContext.Provider value={{ state, dispatch }}>
          <HostRoundWinnerScreen />
        </HostContext.Provider>,
      );

      jest.advanceTimersByTime(3000);

      expect(dispatch).toHaveBeenCalledWith({
        type: 'REMOVE_SUBMITTED_CARDS_FROM_PLAYER',
        payload: {},
      });

      expect(dispatch).toHaveBeenCalledWith({
        type: 'SET_NEXT_CZAR',
        payload: {},
      });

      expect(dispatch).toHaveBeenCalledWith({
        type: 'SELECT_BLACK_CARD',
        payload: {},
      });

      expect(dispatch).toHaveBeenCalledWith({
        type: 'DEAL_WHITE_CARDS',
        payload: {},
      });

      expect(dispatch).toHaveBeenCalledTimes(6);
    });

    it('if max score is met by a player, dispatches GAME_OVER event once the setTimeout executes', () => {
      jest.useFakeTimers();

      const newState = {
        ...state,
        czarSelection: 'ID3',
        players: {
          ...state.players,
          ID3: {
            ...state.players.ID3,
            score: state.gameSettings.winningScore,
          },
        },
      };
      render(
        <HostContext.Provider value={{ state: newState, dispatch }}>
          <HostRoundWinnerScreen />
        </HostContext.Provider>,
      );

      // Dispatch is called with End of Round event
      expect(dispatch).toHaveBeenCalledTimes(1);

      jest.advanceTimersByTime(3000);

      // Dispatch is called with Game Over event
      expect(dispatch).toHaveBeenCalledTimes(2);
      expect(dispatch).toHaveBeenLastCalledWith({
        type: 'GAME_OVER',
        payload: { gameWinner: 'ID3', playerIDs: newState.playerIDs },
      });
    });

    it('if max score is NOT met by the round winning player, GAME_OVER is not dispatched', () => {
      jest.useFakeTimers();

      const newState = {
        ...state,
        czarSelection: 'ID3',
        players: {
          ...state.players,
          ID3: {
            ...state.players.ID3,
            score: state.gameSettings.winningScore - 1,
          },
        },
      };
      render(
        <HostContext.Provider value={{ state: newState, dispatch }}>
          <HostRoundWinnerScreen />
        </HostContext.Provider>,
      );

      // Dispatch is called with End of Round event
      expect(dispatch).toHaveBeenCalledTimes(1);

      jest.advanceTimersByTime(3000);

      expect(dispatch).not.toHaveBeenCalledWith({
        type: 'GAME_OVER',
        payload: { gameWinner: 'ID3', playerIDs: newState.playerIDs },
      });
    });

    it('if max score is met by the round winning player, GAME_OVER is dispatched only after timer is up', () => {
      jest.useFakeTimers();

      const newState = {
        ...state,
        czarSelection: 'ID3',
        players: {
          ...state.players,
          ID3: {
            ...state.players.ID3,
            score: state.gameSettings.winningScore,
          },
        },
      };
      render(
        <HostContext.Provider value={{ state: newState, dispatch }}>
          <HostRoundWinnerScreen />
        </HostContext.Provider>,
      );

      // Dispatch is called with End of Round event
      expect(dispatch).toHaveBeenCalledTimes(1);

      jest.advanceTimersByTime(2999);

      expect(dispatch).toHaveBeenCalledTimes(1);
      expect(dispatch).not.toHaveBeenCalledWith({
        type: 'GAME_OVER',
        payload: { gameWinner: 'ID3', playerIDs: newState.playerIDs },
      });

      jest.advanceTimersByTime(1);

      expect(dispatch).toHaveBeenCalledTimes(2);
      expect(dispatch).toHaveBeenCalledWith({
        type: 'GAME_OVER',
        payload: { gameWinner: 'ID3', playerIDs: newState.playerIDs },
      });
    });
  });
});
