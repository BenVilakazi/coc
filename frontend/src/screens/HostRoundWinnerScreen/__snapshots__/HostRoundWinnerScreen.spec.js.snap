// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Host Winner Screen rendering renders 1`] = `
.c1 {
  background-image: url(inactiveHamburger.svg);
  background-size: cover;
  background-color: transparent;
  border: 0;
  width: 25px;
  height: 25px;
  cursor: pointer;
}

.c0 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .hamburger-container {
  position: absolute;
  right: 17px;
  top: 11px;
  z-index: 10100;
}

.c0 .host-layout-header {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  height: 50px;
}

.c0 .host-layout-header h2 {
  position: relative;
  bottom: -1px;
  font-size: 1rem;
  line-height: 0.8rem;
}

.c0 .components {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
}

.c0 .components .left {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 200px;
  background-color: var(--primary-background-color);
  box-shadow: 0px 4px 12px rgba(0,0,0,0.25);
}

.c0 .components .right {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  position: relative;
  min-width: 440px;
}

.c2 {
  width: 100%;
  background-color: var(--secondary-background-color);
  color: var(--secondary-text-color);
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 486px;
  padding: 40px;
  background-color: var(--primary-background-color);
  position: relative;
}

.c4 > div:not(:last-child) {
  margin-bottom: 24px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c5 .player-info {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0 24px 0 26px;
  padding: 7px 0;
  border-bottom: 1px solid #ccc;
  border-radius: 0;
  background-color: var(--primary-background-color);
}

.c5 .player-info h1 {
  text-transform: uppercase;
  font-size: 24px;
  font-weight: 900;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 1.5em;
  color: var(--primary-text-color);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5 .player-info span {
  color: var(--primary-text-color);
  font-size: 24px;
  font-weight: 900;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c7 .player-info {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0 12px 0 26px;
  padding: 7px 12px;
  border-bottom: 0;
  border-radius: 4px;
  background-color: var(--secondary-background-color);
}

.c7 .player-info h1 {
  text-transform: uppercase;
  font-size: 24px;
  font-weight: 900;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 1.5em;
  color: var(--secondary-text-color);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7 .player-info span {
  color: var(--secondary-text-color);
  font-size: 24px;
  font-weight: 900;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  line-height: 29px;
}

.c8 .join-code-title {
  font-size: 18px;
  color: var(--primary-text-color);
  background-color: var(--primary-background-color);
  font-weight: 700;
  width: 97px;
  height: 21px;
}

.c8 .join-code {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 56px;
  font-weight: bold;
  color: var(--secondary-text-color);
  background-color: var(--secondary-background-color);
  width: 305px;
  height: 70px;
}

.c10 {
  position: relative;
  width: 390px;
  height: 540px;
  background-color: var(--secondary-background-color);
  color: var(--secondary-text-color);
  border-radius: 18px;
  font-weight: 700;
  font-size: 36px;
  padding: 45px 21px 0 21px;
  box-shadow: 4px 4px 24px rgba(0,0,0,0.3);
  width: 276px;
  height: 363px;
  font-size: 24px;
}

.c10 .pickText {
  font-size: 24px;
  margin-right: 7px;
}

.c10 .layeredCards {
  position: absolute;
  bottom: 18px;
  left: 18px;
}

.c10 .pickCount {
  position: absolute;
  bottom: 8px;
  right: 21px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: var(--primary-text-color);
  background: var(--primary-background-color);
  width: 20px;
  height: 20px;
  border-radius: 50%;
  font-weight: bold;
  font-size: 14px;
}

.c13 {
  position: relative;
  border: 1px solid var(--primary-text-color);
  width: 148px;
  height: 206px;
  background-color: var(--primary-background-color);
  color: var(--primary-text-color);
  border-radius: 7px;
  font-weight: 500;
  font-size: 18px;
  overflow-wrap: break-word;
  line-height: 21px;
  padding: 17px 15px 0 15px;
  box-shadow: 0px 4px 12px rgba(0,0,0,0.25);
}

.c13 .whiteCardText {
  max-height: 150px;
  padding-right: 0;
  overflow-y: auto;
}

.c13 .whiteCardText div {
  padding-bottom: 20px;
}

.c13 .stackedCards {
  position: absolute;
  bottom: 7px;
  left: 11px;
  width: 32px;
}

.c13 .backOfCard {
  font-weight: 600;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
}

.c3 .player-list-wrapper {
  overflow-x: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c3 .buttons-wrapper {
  display: block;
  margin-bottom: 20px;
}

.c3 .bottom-left-wrapper {
  margin: 20px auto 32px;
  position: relative;
  bottom: 0;
  left: 0;
  right: 0;
}

.c3 .join-code-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 .host-pregame-button {
  padding: 0;
  background-color: var(--primary-background-color);
  color: var(--primary-text-color);
  margin: auto auto 12px;
}

.c9 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 .winner-display {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 25px 0;
  line-height: 1em;
  font-size: 56px;
  font-weight: 900;
  text-transform: uppercase;
}

.c9 .winner-display span {
  font-size: 24px;
  text-indent: -20px;
  margin-bottom: -8px;
  line-height: 1.5rem;
}

.c12 {
  display: grid;
  grid-template-columns: repeat(1,auto);
  gap: 20px;
  place-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 40px 0;
  width: 100%;
  max-height: 250px;
}

@media (min-width:680px) {
  .c0 .host-layout-header {
    height: 130px;
  }

  .c0 .components .left {
    min-width: 420px;
  }

  .c0 .components .host-layout-header h2 {
    font-size: 2.5rem;
    margin-bottom: -4px;
    line-height: 2rem;
  }
}

@media (min-width:1600px) {
  .c0 .components .left {
    min-width: 700px;
  }

  .c0 .components .host-layout-header h2 {
    font-size: 3.5rem;
    margin-bottom: -35px;
    line-height: 7rem;
  }
}

@media (min-width:3200px) {
  .c0 .host-layout-header {
    height: 200px;
  }

  .c0 .components .left {
    min-width: 1100px;
  }

  .c0 .components .host-layout-header h2 {
    font-size: 6rem;
    margin-bottom: -25px;
    line-height: 7rem;
  }
}

@media (min-width:1025px) {
  .c13 {
    width: 267px;
    height: 363px;
    font-size: 36px;
    line-height: 42px;
    padding: 30px 27px 0 27px;
  }

  .c13 .whiteCardText {
    max-height: 280px;
    padding-right: 0;
  }

  .c13 .stackedCards {
    width: auto;
  }
}

<div
  className="c0 primary-background"
>
  <div
    className="hamburger-container"
  >
    <button
      className="c1"
      onClick={[Function]}
    />
  </div>
  <div
    className="components"
  >
    <div
      className="left"
    >
      <header
        className="c2 host-layout-header"
      >
        <h2>
          CARDS OF CAROUSAL
        </h2>
      </header>
      <div
        className="c3"
      >
        <div
          className="player-list-wrapper"
        >
          <div
            className="c4"
            data-testid="playerList-container"
          >
            <div
              className="c5"
              data-testid="row-foo"
            >
              <img
                alt="card icon"
                data-testid="icon-foo"
                src="offset-white-card-stack-icon.svg"
                style={
                  Object {
                    "visibility": "visible",
                  }
                }
              />
              <div
                className="player-info"
              >
                <h1>
                  foo
                </h1>
                <div
                  className="c6"
                >
                  <svg
                    data-testid="tally-svg"
                    fill="none"
                    height="30"
                    viewBox="0 0 42 30"
                    width="42"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g />
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="c7"
              data-testid="row-bar"
            >
              <img
                alt="card icon"
                data-testid="icon-bar"
                src="black-card-icon.svg"
                style={
                  Object {
                    "visibility": "visible",
                  }
                }
              />
              <div
                className="player-info"
              >
                <h1>
                  bar
                </h1>
                <div
                  className="c6"
                >
                  <svg
                    data-testid="tally-svg"
                    fill="none"
                    height="30"
                    viewBox="0 0 42 30"
                    width="42"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g />
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="c5"
              data-testid="row-baz"
            >
              <img
                alt="card icon"
                data-testid="icon-baz"
                src="offset-white-card-stack-icon.svg"
                style={
                  Object {
                    "visibility": "visible",
                  }
                }
              />
              <div
                className="player-info"
              >
                <h1>
                  baz
                </h1>
                <div
                  className="c6"
                >
                  <svg
                    data-testid="tally-svg"
                    fill="none"
                    height="30"
                    viewBox="0 0 42 30"
                    width="42"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="bottom-left-wrapper"
        >
          <div
            className="join-code-wrapper"
          >
            <div
              className="c8"
            >
              <p
                className="join-code-title"
              >
                JOIN CODE:
              </p>
              <div
                className="join-code"
                data-testid="join-code"
              >
                <p>
                  ABCD
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="right"
    >
      <div
        className="c9"
      >
        <h1
          className="winner-display"
          data-testid="winner-display"
        >
          <span>
            WINNER:
          </span>
          FOO
        </h1>
        <div
          className="c10"
          data-testid="black-card"
        >
          <span>
            Test Black Card
          </span>
          <div
            className="layeredCards"
          >
            <div
              style={
                Object {
                  "position": "relative",
                }
              }
            >
              <img
                alt="offset black and white card stack"
                data-testid="layerCard"
                src="cards-stack-offset-dark-icon.svg"
              />
            </div>
          </div>
          <div
            className="pickCount"
          >
            <span
              className="pickText"
            >
              PICK
            </span>
            <div
              className="c11"
            >
              1
            </div>
          </div>
        </div>
        <div
          className="c12"
        >
          <div
            className="c13"
            data-testid="white-card"
          >
            <div
              className="whiteCardText"
            >
              <div>
                aaaa
              </div>
            </div>
            <div
              className="stackedCards"
              data-testid="stacked-cards"
            >
              <img
                alt="stacked cards"
                src="cards-stack-offset-light-icon.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
